pkg/operator/certrotation/client_cert_rotation_controller.go

```diff
@@ -94,12 +94,22 @@ func (c CertRotationController) Sync(ctx context.Context, syncCtx factory.SyncCo
 		newCondition.Reason = "RotationError"
 		newCondition.Message = syncErr.Error()
 	}
-	_, updated, updateErr := v1helpers.UpdateStaticPodStatus(ctx, c.OperatorClient, v1helpers.UpdateStaticPodConditionFn(newCondition))
-	if updateErr != nil {
-		return updateErr
-	}
-	if updated && syncErr != nil {
-		syncCtx.Recorder().Warningf("RotationError", newCondition.Message)
+	if c.OperatorClient != nil {
+		// if the user has provided a client, we can report status to the object
+		_, updated, updateErr := v1helpers.UpdateStaticPodStatus(ctx, c.OperatorClient, v1helpers.UpdateStaticPodConditionFn(newCondition))
+		if updateErr != nil {
+			return updateErr
+		}
+		if updated && syncErr != nil {
+			syncCtx.Recorder().Warningf("RotationError", newCondition.Message)
+		}
+	} else {
+		// otherwise, we're just going to log; since we don't have an object on the cluster
```

Comment:
Does hypershift not report its status anyhow?

Reply:
They do, but this code is very tightly coupled to the StaticPodCondition, etc - would you rather see the whole reporting bit factored out as an interface?

Reply:
If we can a reconciliation fail more visible for other operators, I think it would be worth it.